src/CoolFileSystem.cpp

```diff
@@ -994,91 +710,174 @@ bool CoolFileSystem::fileUpdate(String update, const char *path) {
   return (true);
 }
 
+
 /**
- *  CoolFileSystem::incrementsavedData():
- *  This method is provided to increment the
- *  savedData flag
+ *  CoolFileSystem::saveMessageToFile(():
+ *  This method is provided to get the
+ *  savedData flag from the file system
  *
- *  \return true if successful , false otherwise
  *
  */
-bool CoolFileSystem::incrementsavedData() {
-
+bool CoolFileSystem::saveMessageToFile(const char *data) {
+  String lastName = "0";
 #if DEBUG == 1
+  FSInfo fs_info;
 
-  Serial.println(F("Entering CoolFileSystem.incrementsavedData()"));
-
+  if (SPIFFS.info(fs_info) == true) {
+    Serial.print(F("used bytes/total bytes : "));
+    Serial.print(fs_info.usedBytes);
+    Serial.print(F("/"));
+    Serial.print(fs_info.totalBytes);
+    Serial.println();
+  }
 #endif
-  // open file
-  File file = SPIFFS.open("/savedDataFlag.txt", "w");
-  // read file
-  if (!file) {
-    Serial.println(F("failed to open savedDataFlag.txt"));
-
-    return (false);
-  } else {
-    // write to file
-    file.print(this->savedData);
 
-    file.print(" ");
+  int nextLog = lastFileSaved();
+  nextLog++;
+  char nextName[32] = "0";
+  snprintf(nextName, 32, "/log/%ld.json", nextLog);
 
-    file.println(this->linesToSkip);
+#if DEBUG == 1
+  Serial.println(F("number for next filename : "));
+  Serial.println(nextLog);
+  Serial.print(F("Name for next file : "));
+  Serial.println(nextName);
+#endif
 
-    file.close();
+  if (!SPIFFS.exists(nextName))
+  {
+    File f = SPIFFS.open(nextName, "w");
+    f.print(data);
+    f.close();
+    Serial.print(F("File Saved as : "));
+    Serial.println(nextName);
+  } else return false;
 
 #if DEBUG == 1
+  if (SPIFFS.info(fs_info) == true) {
+    Serial.print(F("used bytes/total bytes : "));
+    Serial.print(fs_info.usedBytes);
+    Serial.print(F("/"));
+    Serial.print(fs_info.totalBytes);
+    Serial.println();
+  }
+#endif
 
-    Serial.print(F("number of lines to read :"));
-    Serial.println(this->savedData);
-
-    Serial.print(F("number of lines to skip :"));
-    Serial.println(this->linesToSkip);
+  return true;
+}
 
-#endif
 
-    return (true);
+/**
+ *  CoolFileSystem::getsavedData():
+ *  This method is provided to get the
+ *  savedData flag from the file system
+ *
+ *
+ */
+bool CoolFileSystem::isFileSaved() {
+  Dir dir = SPIFFS.openDir("/log");
+  if (dir.next()) {
+    return true;
   }
+  else return false;
 }
+
+
 /**
  *  CoolFileSystem::getsavedData():
  *  This method is provided to get the
  *  savedData flag from the file system
  *
  *
  */
-void CoolFileSystem::getsavedData() {
+int CoolFileSystem::lastFileSaved() {
+  String lastName = "0";
+  String index = "0";
+  int next = 0;
+  int temp = 0;
+  Dir dir = SPIFFS.openDir("/log");
+  //skip through the "directory " 
+  while (dir.next()) {
+    lastName = dir.fileName();
+    index = lastName.substring(5);
+    temp = index.toInt();
```

Comment:
why not 
```
#define JSON_FILE_EXT_SIZE 5
[...]
temp = dir.fileName().substring(JSON_FILE_EXT_SIZE).toInt();
[...]
```
The calls to `print(ln)` below also would need to be changed accordingly

Note: took me some time to figure out 5 was the size of the `.json` extension, hence why I donc like magic number

This is by no means a matter of personal taste but a [known programming antipattern](https://refactoring.guru/replace-magic-number-with-symbolic-constant), excerpt: 

> A magic number is a numeric value that is encountered in the source but has no obvious meaning. This "anti-pattern" makes it harder to understand the program and refactor the code.
> Yet more difficulties arise when you need to change this magic number. Find and replace won't work for this: the same number may be used for different purposes in different places, meaning that you will have to verify every line of code that uses this number.